src/blocks/sandbox-image/imports/index.mjs

```diff
@@ -0,0 +1,324 @@
+		}
+	}
+
+	// Is an image set already?
+	const hasImage = imgObj ? true : false;
+
+	/**
+	 * If there is no image set, and the user can't edit the image show placeholder.
+	 *
+	 * @see https://developer.wordpress.org/block-editor/reference-guides/components/Placeholder/
+	 */
+	if ( ! hasImage && ! canEditImage ) {
+		return (
+			<Placeholder
+				// Required Props - none
+
+				// Customizable Props
+				className="bu-components-image-placeholder"
+				label={ label }
+				instructions={ instructions }
+
+				// @todo decide if any of these or other props should be defined or malleable
+			/>
+		);
+	}
+
+	/**
+	 * If there is no image set, but the user can edit the image, show Media Placeholder.
+	 *
+	 * @see https://github.com/WordPress/gutenberg/blob/HEAD/packages/block-editor/src/components/media-placeholder/README.md
+	 */
+	if ( ! hasImage ) {
+		return (
+			<MediaPlaceholder
+				// Required Props
+				onSelect={ ( media ) => setMediaId( media.id ) }
+				// Customizable Props
+				className="bu-components-image-mediaplaceholder"
+				labels={ labels }
+				allowedTypes={ [ 'image' ] }
+				accept="image/*"
+
+				// @todo decide if any of these or other props should be defined or malleable
+			/>
+		);
+	}
+
+	/**
+	 * Build the alt text attribute.
+	 */
+	let altText = '';
+	if ( altSource === 'alt' ) {
+		altText = imgObj.alt;
+	} else if ( altSource === 'caption' ) {
+		altText = imgObj.caption;
+	} else if ( altSource === 'title' ) {
+		altText = imgObj.title;
+	} else if ( altSource === 'description' ) {
+		altText = imgObj.description;
+	} else {
+		altText = altSource;
+	}
+
+	/**
+	 * Build the image sources, if defined.
+	 *
+	 * @see https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Elements/source#sizes
+	 */
+	if ( debug ) {
+		// eslint-disable-next-line no-console
+		console.log( 'sourceSizes: ', sourceSizes );
+	}
+	const sourcesTag = []; // tags for picture
+	let srcsetAttribute = ''; // attribute for img
+	const srcsetAttributeArray = []; // holder for srcsetAttribute before using join
+	//
+	sourceSizes.forEach( function ( imgSource ) {
+		const imgSourceObj = getImageData( mediaObj, imgSource.srcset );
+		if ( debug ) {
+			// eslint-disable-next-line no-console
+			console.log( 'imgSource: ', imgSource );
+			// eslint-disable-next-line no-console
+			console.log( 'imgSourceObj: ', imgSourceObj );
+		}
+		if ( imgSourceObj ) {
+			// for picture
+			sourcesTag.push(
+				<source
+					srcSet={ imgSourceObj.src }
+					media={ imgSource.media }
+					type={ imgObj.mime_type }
+				/>
+			);
+			// for imgs
+			srcsetAttributeArray.push(
+				`${ imgSourceObj.src } ${ imgSource.descriptor }`.trim()
+			);
+		}
+	} );
+	srcsetAttribute = srcsetAttributeArray.join( ', ' );
+	if ( debug ) {
+		// eslint-disable-next-line no-console
+		console.log( 'sourcesTag: ', sourcesTag );
+		// eslint-disable-next-line no-console
+		console.log( 'srcsetAttribute: ', srcsetAttribute );
+	}
```

Comment:
Could some of this move into block-imports like `getImageData()`? Maybe a small component that returns a set of <source> tags? Or even just into a separate file in this Image component for now to make this index file cleaner.